### DISCUSSION TITLE
Does the new 2.2.* release bring any API/ABI incompatibilities?

### DISCUSSION THREAD

**Discussion by ljavorsk** [zlib-ng/zlib-ng#1796] [Release Feedback]
I reviewed the release notes for each 2.2.* version and didn’t find any changes that might affect the API/ABI used in the 2.1.* version.

Could you please confirm if there are any cases I might have overlooked?

I'm asking this because we plan to add this version to Fedora Rawhide [1] and RHEL-10 [2].
Thank you so much for the response.

[1] https://fedoraproject.org/wiki/Changes/ZlibNG-2.2
[2] https://issues.redhat.com/browse/RHEL-44348

**Comment by Dead2** [comment]
The only change potentially affecting other applications is that we now allocate memory using a single alloc call instead of multiple, and we allocate slightly more (some padding).

Personally I only know of one application that cares about that, namely Nginx, since it pre-allocates memory in order to reduce the number of zlib calls to system alloc functions. Nginx no longer needs to do so, since we do it ourselves now, I am going to post a patch for that in our patches repo, thanks for reminding me.

**Comment by mtl1979** [comment]
We bave script in CI that checks for any ABI breakages. As far as I know, the base ABI files have not been changed/updated in a long time.
